app/src/services/thread/thread.ts

```diff
@@ -1,190 +1,178 @@
 }) => {
-  try {
-    const req = await axios({
-      method: "post",
-      url: `/api/threads/${threadId}/likes`,
-      data: { title },
-    });
-    if (req.status && req.data.threadLikeDocument._id) {
-      onSuccess(req.data.threadLikeDocument._id);
-    } else {
-      throw Error(
-        `Unable to get _id from req.data.threadLikeDocument: ${JSON.stringify(
-          req.data.threadLikeDocument
-        )}`
-      );
-    }
-  } catch (error) {
-    console.error(error);
-    typeof error?.message === "string" &&
-      onError(
-        "Sorry, we're unable to add your reaction at this time, please try again later"
-      );
+  const res = await axios({
+    method: "post",
+    url: `/api/threads/${threadId}/likes`,
+    data: { title },
+  });
+  if (res?.data?.updatedThread) {
+    const processedThreadData = await processThread(res.data.updatedThread);
+    return processedThreadData;
+  } else {
+    throw Error(res.statusText);
   }
 };
 
 const removeThreadReaction = async ({
   threadId,
   threadLikeId,
-  onSuccess,
-  onError,
 }: {
   threadId: string;
   threadLikeId: string;
-  onSuccess: (clearReaction: string | false) => void;
-  onError: (message: string) => void;
 }) => {
-  try {
-    const req = await axios({
-      method: "delete",
-      url: `/api/threads/${threadId}/likes`,
-      data: { threadLikeId },
-    });
-    if (req.status === 200) {
-      onSuccess(false);
-    } else {
-      onError(req.statusText);
-    }
-  } catch (error) {
-    console.error(error);
-    typeof error?.message === "string" &&
-      onError(
-        "Sorry, we're unable to remove your reaction at this time, please try again later"
-      );
+  const res = await axios({
+    method: "delete",
+    url: `/api/threads/${threadId}/likes`,
+    data: { threadLikeId },
+  });
+  if (res?.data?.updatedThread) {
+    const processedThreadData = await processThread(res.data.updatedThread);
+    return processedThreadData;
+  } else {
+    throw Error(res.statusText);
   }
 };
 
 const addComment = async ({
   threadId,
   data,
-  onSuccess,
-  onError,
 }: {
   threadId: string;
   data: { content: string };
-  onSuccess: (data: IThreadComment) => void;
-  onError: (message: string) => void;
 }) => {
-  try {
-    const req = await axios({
-      method: "post",
-      url: `/api/threads/${threadId}/comments`,
-      data,
-    });
-    if (req.status === 200) {
-      onSuccess(req.data.newComment);
-    } else {
-      onError(req.statusText);
-    }
-  } catch (error) {
-    console.error(error);
-    typeof error?.message === "string" &&
-      onError(
-        "Sorry, we're unable to add your post at this time, please try again later"
-      );
+  const res = await axios({
+    method: "post",
+    url: `/api/threads/${threadId}/comments`,
+    data,
+  });
+  if (res?.data?.updatedThread) {
+    const processedThreadData = await processThread(res.data.updatedThread);
+    return processedThreadData;
```

Comment:
```suggestion
   return await processThread(res.data.updatedThread);
```
Are we going to be doing more processing with this variable?

Reply:
I prefer to assign to something on these, when debugging I'll often want to console.log the awaited data.